pkg/rekor/identity.go

```diff
@@ -105,8 +109,215 @@ func (e *IdentityEntry) String() string {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDSourceRepositoryIdentifier,
+			ExtensionValues:  e.SourceRepositoryIdentifier,
+		})
+	}
+	if len(e.SourceRepositoryOwnerURI) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDSourceRepositoryOwnerURI,
+			ExtensionValues:  e.SourceRepositoryOwnerURI,
+		})
+	}
+	if len(e.SourceRepositoryOwnerIdentifier) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDSourceRepositoryOwnerIdentifier,
+			ExtensionValues:  e.SourceRepositoryOwnerIdentifier,
+		})
+	}
+	if len(e.BuildConfigURI) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDBuildConfigURI,
+			ExtensionValues:  e.BuildConfigURI,
+		})
+	}
+	if len(e.BuildConfigDigest) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDBuildConfigDigest,
+			ExtensionValues:  e.BuildConfigDigest,
+		})
+	}
+	if len(e.BuildTrigger) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDBuildTrigger,
+			ExtensionValues:  e.BuildTrigger,
+		})
+	}
+	if len(e.RunInvocationURI) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDRunInvocationURI,
+			ExtensionValues:  e.RunInvocationURI,
+		})
+	}
+	if len(e.SourceRepositoryVisibilityAtSigning) != 0 {
+		exts = append(exts, OIDMatcher{
+			ObjectIdentifier: extensions.OIDSourceRepositoryVisibilityAtSigning,
+			ExtensionValues:  e.SourceRepositoryVisibilityAtSigning,
+		})
+	}
+
+	return exts, nil
+}
+
+// parseObjectIdentifier parses a string representing an ObjectIdentifier in dot notation
+// and converts it into an asn1.ObjectIdentiifer.
+func parseObjectIdentifier(oid string) (asn1.ObjectIdentifier, error) {
+	if len(oid) == 0 {
+		return nil, errors.New("could not parse object identifier: empty input")
+	}
+	nodes := strings.Split(oid, ".")
+	objectIdentifier := make([]int, len(nodes))
+	for i, node := range nodes {
+		if strings.TrimSpace(node) == "" {
+			return nil, errors.New("could not parse object identifier: no characters between two dots")
+		}
+		intNode, err := strconv.Atoi(node)
+		if err != nil {
+			return nil, err
+		}
+		objectIdentifier[i] = intNode
+	}
+	return asn1.ObjectIdentifier(objectIdentifier), nil
+}
+
+// mergeOIDMatchers groups all OID matchers from OIDMatchers, FulcioExtensions, and CustomOIDs into one slice of OIDMatchers
+func mergeOIDMatchers(mvs MonitoredValues) ([]OIDMatcher, error) {
+	fulcioOIDMatchers, err := renderFulcioOIDMatchers(mvs.FulcioExtensions)
+	if err != nil {
+		return nil, fmt.Errorf("error rendering OID Matchers from Fulcio OID extensions: %w", err)
+	}
+	// map of all OID extensions to all associated matching extension values
+	oidMap := make(map[string]map[string]bool)
+	// dedup OID extensions and associated values through one mapping
+	for _, oidMatcher := range mvs.OIDMatchers {
+		oidMap[oidMatcher.ObjectIdentifier.String()] = make(map[string]bool)
```

Comment:
Nit: let's save `oidMatcher.ObjectIdentifier.String()` to a string variable (`oidAsString`?) instead of computing it twice (283,285, then 289,291)